proposals/0285-no-implicit-binds.rst

```diff
@@ -194,6 +194,24 @@ This proposal applies to all alike:
 
     instance forall t. Eq t => C t where -- OK
 
+#. ::
+
+
+    {-# LANGUAGE NoImplicitForAll #-}
+    -- {-# LANGUAGE NoPatternSignatureBinds #-} -- Does not matter whether enabled or disabled
+
+    {-# RULES "name" forall (x :: Maybe a). foo x = 5 #-} -- error: `a` is not bound
+
+   This could be rewritten as::
+
+    {-# LANGUAGE NoImplicitForAll #-}
+    -- {-# LANGUAGE NoPatternSignatureBinds #-} -- Does not matter whether enabled or disabled
+
+    {-# RULES "name" forall a. forall (x :: Maybe a). foo x = 5 #-}
+
+   (The double-\ ``forall`` syntax separates type variables like ``a`` from term variables like ``x``.
+   This syntax is fairly obscure, but does in fact already exist.)
```

Comment:
"... does in fact already exist".   Can you say where it already exists, and give a link to the user manual entry that describes it?

Reply:
I did some digging and it came from https://github.com/ghc-proposals/ghc-proposals/pull/55. (Wow, I didn't realize it's been enough years of having GHC proposals that stuff I'd need to dig up to remember would relate to them!)

It was implemented in [`512eeb9bb9a81e915bfab25ca16bc87c62252064`](https://gitlab.haskell.org/ghc/ghc/-/commit/512eeb9bb9a81e915bfab25ca16bc87c62252064).

It is documented in https://downloads.haskell.org/ghc/latest/docs/html/users_guide/exts/rewrite_rules.html#syntax . If you search for `map/id` you will get the exact spot.

-----

What of this information would you like contained within this PR proper?

Reply:
I was thinking just "...does is alreasdy part of the syntax for RULES (see the 6.19.1.1  "Rewrite rules, Syntax" <link>)."

Reply:
```suggestion
   This syntax is alreasdy part of the syntax for ``RULES``, see `section 6.19.1.1 "Rewrite rules, Syntax" <https://downloads.haskell.org/ghc/latest/docs/html/users_guide/exts/rewrite_rules.html#syntax>`_ of the Users Guide.)
```